Subcommands doesn't work on windows

The subcommand binary isn't found on windows environments

```
C:\> tool config get option
C:\Users\jduque\AppData\Roaming\npm\node_modules\tool\node_modules\args\index.js:342
throw err
^

Error: spawn tool.js-config ENOENT
```

@ntwcklng I tested your solution but still not luck. This is what I get now:

```
C:\> mytool config get option
module.js:471
  throw err
  ^

Error: Cannot find module 'C:\Users\gioyik\Documents\MyProject\mytool-config.js'
```

The `package.json` of `mytool` I have:

```js
  "bin": {
    "mytool": "./bin/mytool.js",
    "mytool-config": "./bin/mytool-config.js"
  },
```

Seems it's looking for `mytool-config.js` file inside the project/folder I run `mytool`.  Hope this helps to understand what is happening.

--- a/lib/utils.js
+++ b/lib/utils.js
@@ -1,6 +1,7 @@
 'use strict';
 
 const spawn = require('child_process').spawn;
+const path = require('path');
 const pkginfo = require('pkginfo');
 const camelcase = require('camelcase');
 const stringSimilarity = require('string-similarity');
@@ -275,10 +276,10 @@ module.exports = {
     }
 
     // Generate full name of binary
-    const full =
-      this.binary +
-      '-' +
-      (Array.isArray(details.usage) ? details.usage[0] : details.usage);
+    const subCommand = Array.isArray(details.usage)
+      ? details.usage[0]
+      : details.usage;
+    let full = this.binary + '-' + subCommand;
 
     const args = process.argv;
     let i = 0;
@@ -288,10 +289,22 @@ module.exports = {
       i++;
     }
 
-    // Run binary of sub command
-    this.child = spawn(full, args, {
-      stdio: 'inherit'
-    });
+    if (process.platform === 'win32') {
+      const binaryExt = path.extname(this.binary);
+      if (path.extname(this.binary)) {
+        full = `${this.binary.replace(binaryExt, '')}-${subCommand}${binaryExt}`;
+      }
+      // Run binary of sub command on windows
+      args.unshift(full);
+      this.child = spawn(process.execPath, args, {
+        stdio: 'inherit'
+      });
+    } else {
+      // Run binary of sub command
+      this.child = spawn(full, args, {
+        stdio: 'inherit'
+      });
+    }
 
     // Throw an error if something fails within that binary
     this.child.on('error', err => {

--- a/lib/utils.js
+++ b/lib/utils.js
@@ -217,9 +217,10 @@ module.exports = {
             initial = items[item].init;
           }
 
-          usage += initial && isVersion === -1
-            ? ' ' + this.handleType(initial)[0]
-            : '';
+          usage +=
+            initial && isVersion === -1
+              ? ' ' + this.handleType(initial)[0]
+              : '';
         }
       }
 
@@ -245,7 +246,9 @@ module.exports = {
       // Add some space around it as well
       if (typeof defaultValue !== 'undefined') {
         if (typeof defaultValue === 'boolean') {
-          description += ` (${defaultValue ? 'enabled' : 'disabled'} by default)`;
+          description += ` (${defaultValue
+            ? 'enabled'
+            : 'disabled'} by default)`;
         } else {
           description += ` (defaults to ${JSON.stringify(defaultValue)})`;
         }
@@ -291,8 +294,12 @@ module.exports = {
 
     if (process.platform === 'win32') {
       const binaryExt = path.extname(this.binary);
+      const mainModule = process.mainModule.filename;
+
+      full = `${mainModule}-${subCommand}`;
+
       if (path.extname(this.binary)) {
-        full = `${this.binary.replace(binaryExt, '')}-${subCommand}${binaryExt}`;
+        full = `${mainModule.replace(binaryExt, '')}-${subCommand}${binaryExt}`;
       }
       // Run binary of sub command on windows
       args.unshift(full);

